### PR TITLE
[s] Fixes ore satchel exploits, various storage size circumventions.

### LINKED ISSUE #37260: Mining satchels pickup ore while in backpack
Server revision compiled on: 2018-04-15

The following pull requests are currently test merged:
37024: 'Puts some brakes on the Cult Train' by Robustin at commit a2366c
37039: 'testing new icons for webs!' by Armhulen at commit 5ad857
36754: '[s] Invisimin HUD Fix + Changeling AI HUD Fix + Fixes Changeling Digital Camo with AI' by Davidj361 at commit 69a66b
37202: 'Fixes cat parts not showing up in perfs.' by DaxDupont at commit eef589

@kevinz000

### LINKED ISSUE #43018: Mining sachel of holding full, destroying ore on "pickup"
Issue reported from  Round ID: 103031 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

For some reason, my mining sachel of holding says it is full. There is nothing in it. I still pick up ore from the ground, but it is instantly destroyed. I do not know what has caused this.

### LINKED ISSUE #43077: Satchels of Holding Dump Ores into Backpack Instead of Satchel of Holding
Issue reported from  Round ID: 103342 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

Title. Also, they say that they're full and can't hold anymore when they clearly can. It puts the ores into your hand if that is empty, as well.

Ayla Storm fires the advanced plasma cutter!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
Your mining satchel is full and can't hold any more!
You put the advanced plasma cutter into the leather satchel.
You put the iron ore into the mining satchel of holding.
You put the silver ore into the mining satchel of holding.
Ayla Storm fires the proto-kinetic accelerator!
Your mining satchel is full and can't hold any more!
You put the iron ore into the mining satchel of holding.
An eerie moan rises on the wind. Sheets of burning ash blacken the horizon. Seek shelter.
Your mining satchel is full and can't hold any more!
scrnshot1
scrnshot2
scrnshot3

Reproduce:
1) Get a satchel of holding and a backpack. Put a normal mining satchel in your backpack. Wear the backpack, stick the satchel in your pocket or hold it.
2) Go mining.
3) If your hands are full, when you walk over an ore, it will deposit into your backpack and say the satchel is full. If your hands aren't full, it'll deposit into your hands and say the satchel is full. If you put the ores into the satchel, the satchel will accept them, so it clearly isn't full.

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Fixes #43077
Fixes #43018
Fixes #37260

Stops the starter mining satchel from picking up ores from backpack, and prevents the overflowing ores from getting dropped in backpack giving you infinite ore storage. Also fixes a couple other (easier) ways of exploiting this that I will not disclose. 

As a bonus, tidies up some other stuff dropping inside storage regardless if there is room or not. Notably .357 shells when using a speedloader on a bagged gun. I skimmed the usages of drop_location(), and could not find a single legitimate reason to AllowDrop() into storage, and it is an unsafe proc to use for the purpose anyway. If you come across any, let me know.

## Why It's Good For The Game
Bugfixes

## Changelog
:cl: Skoglol
fix: Mining bags will no longer drop ore into backpack.
fix: Mining bags in backpack no longer interferes with other mining bags.
fix: .357 casings now always drops on floor when speedloaded.
fix: Fixes some storage size circumventions.
/:cl:

### CODE REVIEW

**Comment by Barhandar** [comment]
Allow bluespace satchels to work from anywhere, because bluespace magic and they're not capable of overflow.

**Comment by kriskog** [comment]
Satchels gathering while in backpack wouldn't cause overflow after this PR. It's not as easy as just adding backpack as a valid slot though, or you'd just cause the teleporting issue again. Either way, it is out of scope for this PR.

File: code/modules/jobs/job_types/shaft_miner.dm [hunk 1]
```diff
@@ -28,9 +28,9 @@
 	gloves = /obj/item/clothing/gloves/color/black
 	uniform = /obj/item/clothing/under/rank/miner/lavaland
 	l_pocket = /obj/item/reagent_containers/hypospray/medipen/survival
-	r_pocket = /obj/item/flashlight/seclite
+	r_pocket = /obj/item/storage/bag/ore	//causes issues if spawned in backpack
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/jobs/job_types/shaft_miner.dm]: can you explain?

Comment (kriskog) [reply code/modules/jobs/job_types/shaft_miner.dm]: Sure. The bag relies on a mobhook component which is pre-filled when it's spawned in bag. I didn't bother digging too deep, but I assume when you spawn in with gear, /equipped() is called. If you just leave the bag in your backpack, the bag behaves as if it is in your belt/pocket slot - regardless of where the backpack is.
Spawning it in your pocket/belt instead forces you to call /dropped() on it if you put it somewhere else, deleting the mobhook.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/jobs/job_types/shaft_miner.dm]: cringe

**Pull request merged**